src/deserialize.zig

```diff
@@ -37,10 +37,14 @@ inline fn safeReadSliceIntBig(comptime T: type, payload: []const u8, out: *T) vo
 
 // Returns the size of the payload as well as the offset to the
 // start of the actual data.
-fn sizeAndDataOffset(payload: []const u8) struct { size: usize, offset: usize } {
+fn sizeAndDataOffset(payload: []const u8) !struct { size: usize, offset: usize } {
```

Comment:
I had to return an error here, which is actually a good thing: going over the encoded payload once requires more _a priori_ checks than before.